libs/blocks/global-navigation/features/breadcrumbs/breadcrumbs.js

```diff
@@ -0,0 +1,108 @@
+import { getMetadata } from '../../../../utils/utils.js';
+import { toFragment, lanaLog } from '../../utilities/utilities.js';
+
+const metadata = {
+  seo: 'breadcrumb-seo',
+  fromFile: 'breadcrumbs-from-file',
+  hideCurrent: 'breadcrumbs-hide-current-page',
+  hiddenEntries: 'breadcrumbs-hidden-entries',
+  pageTitle: 'breadcrumbs-page-title',
+  base: 'breadcrumbs-base',
+  fromUrl: 'breadcrumbs-from-url',
+};
+
+const setBreadcrumbSEO = (breadcrumb) => {
+  const seoDisabled = getMetadata(metadata.seo) === 'off';
+  if (seoDisabled || !breadcrumb) return;
+  const breadcrumbSEO = {
+    '@context': 'https://schema.org',
+    '@type': 'BreadcrumbList',
+    itemListElement: [],
+  };
+  const items = breadcrumb.querySelectorAll('ul > li');
+  items.forEach((item, idx) => {
+    const link = item.querySelector('a');
+    breadcrumbSEO.itemListElement.push({
+      '@type': 'ListItem',
+      position: idx + 1,
+      name: link ? link.innerHTML : item.innerHTML,
+      item: link?.href,
+    });
+  });
+  const script = toFragment`<script type="application/ld+json">${JSON.stringify(
+    breadcrumbSEO,
+  )}</script>`;
+  document.head.append(script);
+};
+
+const removeHiddenEntries = ({ ul }) => {
```

Comment:
The operation to remove entries is super cheap as it's not appended to the dom yet (the action costs ~1ms, even with x6 CPU throttle) 

We need to do this for the authored breadcrumbs base anyway, so I'd prefer a consistent way rather than more code for the automatic from url one

Reply:
thanks for looking into this, I'm fine with the results.